docs/nomad_simulations.md

```diff
@@ -1 +1,499 @@
+
+    * An `ElectronicBandGap` whose value is 2 eV.
+    * An `ElectronicBandGap` varying with `Temperature`, whose values are `[1, 1.5, 2]` in eV for `[100, 150, 200]` temperatures in Kelvin.
+
+    In the second situation of an electronic band gap which varies with the `Temperature`, what happens if you directly assign `value` **before** defining the `Temperature` variables sub-section in `ElectronicBandGap`?
+
+
+??? success "Solution 2.8"
+    We can import the `ElectronicBandGap` property and assign the first case of an electronic band gap which is 2 eV:
+    ```python
+    from nomad_simulations.schema_packages.properties import ElectronicBandGap
+    band_gap = ElectronicBandGap()
+    band_gap.value = 2 * ureg.eV
+    scf_outputs.electronic_band_gaps.append(band_gap)
+    ```
+
+    We can do the same for the temperature-dependent band gap by importing and defining the `Temperature` variable sub-section (note that `variables` is a repeated sub-section, thus we need to assign a list):
+    ```python
+    from nomad_simulations.schema_packages.variables import Temperature
+    band_gap_T = ElectronicBandGap(variables=[Temperature(points=[100, 150, 200] * ureg.kelvin)])
+    band_gap_T.value = [1, 1.5, 2] * ureg.eV
+    scf_outputs.electronic_band_gaps.append(band_gap_T)
+    ```
+
+    Here, we have assigned first `variables` before the `value` of the property. However, if we want to assign first the `value`:
+    ```python
+    band_gap_T = ElectronicBandGap()
+    band_gap_T.value = [1, 1.5, 2] * ureg.eV
+    ```
+    we will get a `ValueError` due to the fact that `variables` is not set and the shape of `value` is not empty:
+    ```sh
+    ValueError: The shape of the stored `value` [3] does not match the full shape [] extracted from the variables `n_points` and the `shape` defined in `PhysicalProperty`.
+    ```
+
+    Thus, the `variables` sub-sections **must** be set before setting the `value` of a physical property. This is because the `class PhysicalProperty` is doing validations on the shape of the `variables` and `value`, and it only works if `variables` is set first. Only when the shape of `value` is empty and due to the fact that `variables` is set by default to an empty list, then we would not get any error.
+
+
+## Extra: The `normalize()` class function {#normalize-function}
+
+Each base section defined using the NOMAD schema has a set of public functions which can be used at any moment when reading and parsing files in NOMAD. The `normalize(archive, logger)` function is a special case of such functions, which warrants an in-depth description.
+
+This function is run within the NOMAD infrastructure by the [`MetainfoNormalizer`](https://github.com/nomad-coe/nomad/blob/develop/nomad/normalizing/metainfo.py) in the following order:
+
+1. A child section's `normalize()` function is run before their/its parents' `normalize()` function.
```

Comment:
not sure about the "their/its" usage, are you trying to say there could be one or more children or one or more parents, or both?

Reply:
Should be singular, I am talking about a child section w.r.t multiple parents.